engine/equations.h

```diff
@@ -342,7 +342,7 @@ namespace MathDAG
     /// create a variable DAG. returns cached value if previously called
     NodePtr makeDAG(const string& valueId, const string& name, VariableType::Type type);
     NodePtr makeDAG(VariableBase& v)
-    {v.ensureValueExists(); return makeDAG(v.valueId(),VariableValue::uqName(v.name()),v.type());}
+    {v.ensureValueExists(v.vValue()); return makeDAG(v.valueId(),VariableValue::uqName(v.name()),v.type());}
```

Comment:
Strictly unnecessary to pass anything other than nullptr here and in other places below, except when variables are being renamed. I would have probably made the parameter default to nullptr. But harmless nevertheless (if v.vValue is not null, ensureValueExists is a no operation), so I'll merge the PR.